### PR TITLE
[DPP-1363] Refactor IdPageSizing

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala [hunk 1]
```diff
@@ -18,46 +18,71 @@ case class IdPageSizing(
...
+          s"Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$numOfDecomposedFilters] too large?"
       )
-      IdPageSizing(calculatedMaxIdPageSize, calculatedMaxIdPageSize)
-    } else if (calculatedMaxIdPageSize < maxIdPageSize) {
+      IdPageSizing(calculated, calculated)
+    } else if (calculated < maxIdPageSize) {
       logger.info(
-        s"Calculated maximum ID page size supporting API stream memory limits [$calculatedMaxIdPageSize] is low. Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$filterSize] too large?"
+        s"Calculated maximum ID page size supporting API stream memory limits [$calculated] is low. " +
+          s"Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$numOfDecomposedFilters] too large?"
```
Comment (atriantafyllos-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala:64]: There is a `Warning:` logged in info level. It seems that there is not a situation where the calculation is correct.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala [hunk 2]
```diff
@@ -18,46 +18,71 @@ case class IdPageSizing(
...
+    val calculated = calculateMaxNumOfIdsPerPage(
+      workingMemoryInBytesForIdPages = workingMemoryInBytesForIdPages,
+      numOfDecomposedFilters = numOfDecomposedFilters,
+      numOfPagesInIdPageBuffer = numOfPagesInIdPageBuffer,
+    )
+    // maxNumberOfIdsPerIdPage can override this if it is smaller
+    val minIdPageSize = Math.min(10, maxIdPageSize)
+    // maxNumberOfIdsPerIdPage can override this if it is smaller
+    val recommendedIdPageSize = Math.min(NumOfBtreeLeafPageEntriesApprox, maxIdPageSize)
+    // An id occupies 8 bytes
```
Comment (atriantafyllos-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala:47]: I am not familiar with this code and I do not know why an id occupies 8B. Could you explain to me or add a reference?

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala (hunk 1 above)
Comment (atriantafyllos-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala:64]: i.e. there is a complaint for each branch of the if-else statement.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala]: It's a long which has 8 bytes. Added in the in code comment

Comment (atriantafyllos-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala]: Shouldn't it be in `calculateMaxNumOfIdsPerPage`?

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala:64]: The last one is without a warning.
I intentionally haven't changed the wording of these messages (as e.g. someone might rely on them to diagnose problem). I only refactored the surrounding code

**Review by atriantafyllos-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala]: Correct, fixed!

**Pull request merged**